src/core/geometry/qgslinestring.cpp

```diff
@@ -574,17 +575,34 @@ bool QgsLineString::fromWkb( QgsConstWkbPtr &wkbPtr )
 }
 
 QgsRectangle QgsLineString::calculateBoundingBox() const
+{
+  QgsBox3d b = calculateBoundingBox3d();
```

Comment:
Can you leave the original 2d implementation intact here? I realise it's a bit of duplicate code, but the 2d bounding box calculation is used a **LOT** in QGIS and needs to be as fast as we possibly can make it (in other words we don't want the added cost of the z range calculation here when we're don't explicitly need it)

Reply:
ok!